File: javascript/1_twoSum.js
var twoSum = function (nums, target) {
  if (nums.length < 2) return [];
  let map = new Map();
  for (let i = 0; i < nums.length; i++) {
    const tmpTarget = target - nums[i];
    if (map.has(tmpTarget)) {
      return [map.get(tmpTarget), i];
    } else {
      map.set(nums[i], i)
    }
  }
};

File: go/1_twoSum.go
func twoSum(nums []int, target int) []int {
    m := make(map[int]int)
    for k,v := range nums {
        index, ok := m[target - v]
        if ok {
            return []int{index, k}
        } else {
            m[v] = k
        }
    }
    return []int{}
}